### PR TITLE
LPS-141237 Add test CustomElementCanConfigurePortletName

### PR DESCRIPTION
**Pull request by sergiojimcos**
**Test Scenario:**
- Given: Remote App
- When: change the portlet alias (Name field)
- Then: the portlet alias is reflected on the remote app in the add widget panel

**What is new?**
RemoteApps#CustomElementCanConfigurePortletName: When Custom Element is displayed with a specific name, this name could be changeable and been viewed adding Widget in Page Editor.

**JIRA Ticket:** 
https://issues.liferay.com/browse/LPS-141237

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase [hunk 1]
```diff
@@ -271,6 +271,81 @@ definition {
...
+			ContentPagesNavigator.openEditContentPage(
+				pageName = "Test Page",
+				siteName = "Guest");
+
+			PageEditor.addWidget(portletName = "Vanilla Counter");
+
+			PageEditor.clickPublish();
+
+			Navigator.gotoPage(pageName = "Test Page");
+		}
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/remotewebapps/RemoteApps.testcase:307]: There's a way to add widget to page via JSON. Try JSONLayout.addWidgetToPublicLayout

You'll need to add a new line to the portletId list in the macro file JSONLayoutUtil._generateWidget.

**Review by john-co — CHANGES_REQUESTED** [review]
Looks good overall, just add the JSON refactor and good to forward. Thanks! :)

File: portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutSetter.macro [hunk 2]
```diff
@@ -117,6 +117,13 @@ definition {
 				typeSettings = "${typeSettings}",
 				widgetName = "${widgetName}");
 		}
+		else if (isSet(widgetName) && isSet(remoteAppEntryId)) {
+			var typeSettings = JSONLayoutUtil._addWidgetToTypeSettings(
+				column = "${column}",
+				remoteAppEntryId = "${remoteAppEntryId}",
+				typeSettings = "${typeSettings}",
+				widgetName = "${widgetName}");
+		}
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutSetter.macro:126]: I don't think you need to create a separate else if statement here if we go with 

> Maybe instead, right after _generateWidget, we can check if (isSet(remoteAppEntryId)) and append the remoteAppEntryID.

File: portal-web/test/functional/com/liferay/portalweb/macros/RemoteApps.macro [hunk 3]
```diff
@@ -190,6 +190,21 @@ definition {
 		PortletEntry.publish();
 	}
 
+	macro getRemoteAppEntryIdValue {
+		WaitForSPARefresh();
+
+		RemoteApps.goToRemoteAppsPortlet();
+
+		Click(locator1 = "xPath=(//div[@class='table-list-title']//a)");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/macros/RemoteApps.macro:198]: Why not use something from RemoteApps.path? something like:
```
			Click(
				key_tableEntryName = "Vanilla Counter",
				locator1 = "RemoteApps#TABLE_ENTRY_NAME_REMOTE_TABLE");
```

File: portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutUtil.macro [hunk 4]
```diff
@@ -10,7 +10,14 @@ definition {
 
 		Variables.assertDefined(parameterList = "${typeSettings}");
 
-		var widget = JSONLayoutUtil._generateWidget(widgetName = "${widgetName}");
+		if (isSet(remoteAppEntryId)) {
+			var widget = JSONLayoutUtil._generateWidget(
+				remoteAppEntryId = "${remoteAppEntryId}",
+				widgetName = "${widgetName}");
+		}
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutUtil.macro:17]: I don't quite like adding remoteAppEntryID into _generateWidget. It's expecting 		Variables.assertDefined(parameterList = "${widgetName}"); But doesn't need remoteAppEntryID.

Maybe instead, right after _generateWidget, we can check if (isSet(remoteAppEntryId)) and append the remoteAppEntryID.

**Review by john-co — CHANGES_REQUESTED** [review]

**Comment by john-co** [comment]
ci:test:frontend

**Comment by john-co** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:frontend

**Comment by sergiojimcos** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutUtil.macro [hunk 5]
```diff
@@ -10,13 +10,10 @@ definition {
...
+		var widget = JSONLayoutUtil._generateWidget(widgetName = "${widgetName}");
+
 		if (isSet(remoteAppEntryId)) {
-			var widget = JSONLayoutUtil._generateWidget(
-				remoteAppEntryId = "${remoteAppEntryId}",
-				widgetName = "${widgetName}");
-		}
-		else {
-			var widget = JSONLayoutUtil._generateWidget(widgetName = "${widgetName}");
+			var widget = StringUtil.replace("${widget}", "remoteAppEntryId", "${remoteAppEntryId}");
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/macros/json/layout/JSONLayoutUtil.macro:16]: I think a better method to use is `StringUtil.add`. But looks fine to forward/merge for now and can follow up with refactor later if you get the chance.

**Review by john-co — APPROVED** [review]

**Comment by john-co** [comment]
ci:forward

**Pull request closed without merging**